### PR TITLE
Document the Release Process

### LINKED ISSUE #169: Document the release process
### What is the areas you would like to add the new feature to?

Go-COSE Library

### Is your feature request related to a problem?

We are currently working out of main, where all approved/merged updates are pushed to main. 
As a new release is queued, we stabilize out of main. This has served us well thus far. 
Open question:
- Do we anticipate stabilizing a release, while developing a new version concurrently? This may require a more complex branching strategy.
- Do we believe we can work around our stabilize in main process? 

Either way, we should document the current working process.

### PR DESCRIPTION
**Pull request by SteveLasker**
fixes #169

### CODE REVIEW

File: release-management.md
Comment (OR13) [inline release-management.md:90]:
```suggestion
  - Once no triaged issues or pull requests (prs) are scoped to the release, a release candidate (rc) is cut.
```

File: release-management.md
Comment (OR13) [inline release-management.md:93]: ?

File: release-management.md
Comment (OR13) [inline release-management.md:125]:
```suggestion
```

Comment (OR13) [reply release-management.md:125]: No need for history in a policy doc

**Review by OR13 — APPROVED** [review]
Overall this seems like a good baseline.
We can modify these policies as we encounter reason to, and as we learn from the experience of applying them.

You may consider noting that maintainer will update the release checklist and management policy based on feedback.

File: release-management.md
Comment (SteveLasker) [reply release-management.md:93]: Can I get a wee bit more about which part is the question? 
This section is focused on deleting alphas and rcs, as a forcing function (when needed) to an actual release. 
We know that orgs wind up depending on pre-release builds, and we could break someone by removing a pre-release build, so I'm just putting some process around this.

File: release-management.md
Comment (SteveLasker) [reply release-management.md:103]: same as [above](https://github.com/veraison/go-cose/pull/188#discussion_r1674277901)

File: release-management.md
Comment (SteveLasker) [reply release-management.md:125]: The purpose was to reference the existing security reviews, to provide visibility and confidence in how we manage our policy. We have, and will do security reviews, but will not do them for all releases.
How do we create visibility to reviews, so it doesn't look like we're dismissing the need?

**Comment by SteveLasker** [comment]
> You may consider noting that maintainer will update the release checklist and management policy based on feedback.
👍

File: release-management.md
Comment (yogeshbdeshpande) [inline release-management.md:57]:
```suggestion
The go-cose project will issue minor releases when incremental improvements, or bug fixes are added to existing functionality.
```

File: release-management.md
Comment (yogeshbdeshpande) [inline release-management.md:48]: Line 48 is not clear to me.

New capabilities consumers may want to opt-into.

I do not think, new functionality can be obtained by getting the latest minor release... I am confused what the text is intending to state?

File: release-management.md
Comment (yogeshbdeshpande) [reply release-management.md:93]: I think, the comment is for line 93, What is the context of `-Not supported` ?

File: release-management.md
Comment (yogeshbdeshpande) [inline release-management.md:100]:
```suggestion
  - Release candidate is cut from the branch where the work is done.
```

File: release-management.md
Comment (SteveLasker) [reply release-management.md:93]: I can add a link to the ["support"](https://github.com/veraison/go-cose/pull/188/files#diff-0f9f71fb66352a76730712a4204600abd7a6f46b400b4345886ab61cf30c45b4R105) section

File: release-management.md
Comment (yogeshbdeshpande) [inline release-management.md:12]: Please also state, somewhere in this document, where and how we want the external users to understand what is the current stable `go-cose` release ?

Comment (yogeshbdeshpande) [reply release-management.md:12]: is it through the REAMDE.md or somewhere else?

File: release-management.md
Comment (yogeshbdeshpande) [inline release-management.md:43]: Do we want to maintain all the Release Branches..?

**Review by yogeshbdeshpande — COMMENTED** [review]
I have made few comments for you to have a look...

File: release-management.md
Comment (SteveLasker) [reply release-management.md]: Added a link to the not-supported line. Pease unresolve if it's something else.

File: release-management.md
Comment (SteveLasker) [reply release-management.md:43]: I believe the answer should be yes. Even if something is unsupported, or even has a bug/issue, a consumer may be dependent upon it until they can make changes, or take time to take a newer version.

File: release-management.md
Comment (SteveLasker) [reply release-management.md]: Thanks, Yogesh.
Please take a look at the updated changes.

File: release-management.md
Comment (SteveLasker) [reply release-management.md:12]: I was trying to avoid having to update the readme every time we do a release, and leave it to the [github releases](https://github.com/veraison/go-cose/releases) page. However, the [current readme](https://github.com/veraison/go-cose?tab=readme-ov-file#project-status) does state the current release, so we could continue to update that, or remove it and simply say, check the releases page for the most current.

Do you think we need to update both?

Comment (yogeshbdeshpande) [reply release-management.md:12]: I think, it is sensible to add a permanent reference in README.md where to find the latest `go-cose` release!

**Pull request merged**
